R/decisiontreeutils.R

```diff
@@ -0,0 +1,431 @@
+#' Error checking pre-model
+#' Does not return anything - just throws errror
+#'
+#' @param config list of config options
+#' @param the.data incoming data
+preModelCheckDT <- function(config, the.data) {
+  data_names <- names(the.data)
+  names_list <- getNamesFromOrdered(config$used.weights, data_names)
+  name_y_var <- names_list$y
+  cp <- ifelse(config$the.cp == "Auto" || config$the.cp == "", .00001, config$the.cp)
+
+
+  the_target <- the.data[[name_y_var]]
+  if (is.numeric(the_target) && length(unique(the_target)) < 5 && !is_XDF) {
+    AlteryxMessage("The target variable is numeric, however, it has 4 or fewer unique values.", iType = 2, iPriority = 3)
+  }
+
+  if(rpart_params$cp < 0 || rpart_params$cp > 1) {
+    stop.Alteryx("The complexity parameter must be between 0 and 1. Please try again.")
+  }
+
+  if(is.na(as.numeric(config$the.cp)) && !(config$the.cp == "Auto" || config$the.cp == "")) {
+    stop.Alteryx("The complexity parameter provided is not a number. Please enter a new value and try again.")
+  }
+}
+
+
+#' Creation of components for model object evaluation
+#'
+#' @param config list of config options
+#' @param data list of datastream inputs
+#' @return list with components needed to create model
+createDTParams <- function(config, data) {
+  # use lists to hold params for rpart and rxDTree functions
+  param_list <- list()
+
+  # Determine if a compute context is being used, and whether it is XDF
+  # The mechanism for doing this depends on how the information is transfered
+  xdf_properties <- getXdfProperties("#1")
+  param_list$is_XDF <- xdf_properties$is_XDF
+  param_list$xdf_path <- xdf_properties$xdf_path
+
+  # get data param
+  the.data <- data$data_stream1
+  data_names <- names(the.data)
+  param_list$data <- quote(the.data)
+
+  # Get the field names
+  names_list <- getNamesFromOrdered(config$used.weights, data_names)
+  name_y_var <- names_list$y
+  names_x_vars <- names_list$x
+  name_weight_var <- names_list$w
+
+  # use field names to get formula param
+  param_list$dtree_formula <- makeFormula(names_x_vars, name_y_var)
+
```

Comment:
In that case, I would directly use `names_list$y` rather than `name_y_var`. The more the symbols you end up using in your code, the more difficult it becomes to keep track of them. In this case, I know that `names_list` contains the names I am interested in. In general, I like to avoid creating intermediate variables which are purely assignments (e.g of the kind `x <- y$a`, since it adds overhead of additional symbols to grep.